planner/optimize.go

```diff
@@ -89,7 +90,10 @@ func GetExecuteForUpdateReadIS(node ast.Node, sctx sessionctx.Context) infoschem
 		}
 		if preparedPointer, ok := vars.PreparedStmts[execID]; ok {
 			if preparedObj, ok := preparedPointer.(*core.CachedPrepareStmt); ok && preparedObj.ForUpdateRead {
-				return domain.GetDomain(sctx).InfoSchema()
+				if vars.LocalTemporaryTables != nil {
```

Comment:
You can remove this check `vars.LocalTemporaryTables != nil` because `AttachLocalTemporaryTableInfoSchema` will check it automatically. In fact, we do not expect other codes be aware of `LocalTemporaryTables` in session.